services/QuillLMS/app/models/activity_session_partner_metadata.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: activity_session_partner_metadata
+#
+#  id                  :bigint           not null, primary key
+#  partner             :text             not null
+#  sent_to_partner     :datetime
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#  activity_session_id :integer          not null
+#  external_id         :text             not null
+#
+# Indexes
+#
+#  index_activity_session_partner_metadata_on_activity_session_id  (activity_session_id) UNIQUE
+#  index_activity_session_partner_metadata_on_external_id          (external_id) UNIQUE
+#
+class ActivitySessionPartnerMetadata < ApplicationRecord
+  PARTNERS = [
+    FISHTANK = 'fishtank'
+  ]
+
+  belongs_to :activity_session
+
+  validates :partner, presence: true, inclusion: { in: PARTNERS }
+
+  def send_to_partner
+    send_to_fishtank if partner == FISHTANK
```

Comment:
This is sorta early and could be a later follow-on / refactor, but it might be nice to extract this into a config, e.g. maybe each partner has a service defined.

```ruby
COMPLETION_SERVICES = {
  FISHTANK => PartnerSessions::FishtankCompletion
}

def completion_service = COMPLETION_SERVICES.fetch(partner)
def complete_session
  completion_service.run(self)
  update!(sent_to_partner: DateTime.current)
end